File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    player_indices = {player: idx for idx, player in enumerate(players)}
    for call in callings:
        i = player_indices[call]
        players[i], players[i-1] = players[i-1], players[i]
        player_indices[call] -= 1
        player_indices[players[i]] += 1
        
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        var playersMap = new HashMap<String, Integer>();
        for(int i = 0; i<players.length; i++){
            playersMap.put(players[i], i);
        }
        for(String call : callings){
            int i = playersMap.get(call);
            String tm = players[i-1];
            players[i-1] = players[i];
            players[i] = tm;
            playersMap.put(players[i-1], i-1);
            playersMap.put(players[i], i);
        }
        return players;
    }
}